crates/fsrt/src/forge_project.rs

```diff
@@ -0,0 +1,165 @@
+        iter: I,
+        secret_packages: Vec<PackageData>,
+    ) -> ForgeProject<'_> {
+        let sm = Arc::<SourceMap>::default();
+        let target = EsVersion::latest();
+        let globals = Globals::new();
+        let ctx = AppCtx::new(src);
+        let ctx = iter.into_iter().fold(ctx, |mut ctx, p| {
+            let sourcemap = Arc::clone(&sm);
+            GLOBALS.set(&globals, || {
+                debug!(file = %p.display(), "parsing");
+                let src = self.load_file(p.clone(), sourcemap);
+                debug!("loaded sourcemap");
+                let mut recovered_errors = vec![];
+                let module = parse_file_as_module(
+                    &src,
+                    Syntax::Typescript(TsConfig {
+                        tsx: true,
+                        decorators: true,
+                        ..Default::default()
+                    }),
+                    target,
+                    None,
+                    &mut recovered_errors,
+                )
+                .unwrap();
+                debug!("finished parsing");
+                let mut hygeine = resolver(Mark::new(), Mark::new(), true);
+                let module = module.fold_with(&mut hygeine);
+                ctx.load_module(p, module);
+                ctx
+            })
+        });
+        let keys = ctx.module_ids().collect::<Vec<_>>();
+        debug!(?keys);
+        let env = run_resolver(ctx.modules(), ctx.file_resolver(), secret_packages);
+        ForgeProject {
+            sm,
+            ctx,
+            env,
+            funcs: vec![],
+        }
+    }
+
+    fn get_paths(&self) -> HashSet<PathBuf>;
+
+    fn get_secret_packages(&self) -> Vec<PackageData>;
+
+    fn get_manifest(&self) -> ForgeManifest<'_>;
+}
+pub(crate) struct ForgeProject<'a> {
+    #[allow(dead_code)]
+    pub sm: Arc<SourceMap>,
+    pub ctx: AppCtx,
+    pub env: Environment,
+    pub funcs: Vec<ResolvedEntryPoint<'a>>,
+}
+
+impl<'a> ForgeProject<'a> {
+    #[inline]
+    pub fn add_funcs<I: IntoIterator<Item = Entrypoint<'a, Resolved>>>(&mut self, iter: I) {
+        self.funcs.extend(iter.into_iter().filter_map(|entrypoint| {
+            let (func_name, path) = entrypoint.function.into_func_path();
+            let module = self.ctx.modid_from_path(&path)?;
+            let def_id = self.env.module_export(module, func_name)?;
+            Some(ResolvedEntryPoint {
+                func_name,
+                path,
+                module,
+                def_id,
+                invokable: entrypoint.invokable,
+                webtrigger: entrypoint.web_trigger,
+                admin: entrypoint.admin,
+            })
+        }));
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct ForgeProjectFromDir {
+    #[allow(dead_code)]
+    pub dir: PathBuf,
+    pub manifest_file_content: String,
+}
+
+#[derive(Debug, Clone)]
+pub(crate) struct MockForgeProject<'a> {
+    pub files_name_to_source: HashMap<String, String>,
```

Comment:
```suggestion
    pub files_name_to_source: HashMap<PathBuf, String>,
```

Since, we might want to test non UTF-8 file names in the future.

Reply:
fixed! thank you :)